libs/input/src/index.tsx

```diff
@@ -0,0 +1,74 @@
+import {
+  FormControl,
+  FormErrorMessage,
+  FormHelperText,
+  FormLabel,
+  Input as ChakraInput,
+  InputGroup,
+  InputRightElement,
+} from '@chakra-ui/react';
+import React, { ChangeEvent } from 'react';
+
+import { IconClose } from '@millicast-react/dolbyio-icons';
+
+import { InputProps } from './types';
+
+const Input = ({ disabled, error, label, helper, onChange, placeholder, required, testId, value }: InputProps) => {
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    onChange(event.target.value);
+  };
+
+  const handleClear = () => {
+    onChange('');
+  };
+
+  return (
+    <FormControl isDisabled={disabled} isInvalid={!!error} isRequired={required} test-id={testId}>
+      <FormLabel
+        background="dolbyNeutral.800"
+        fontSize="12px"
+        fontWeight="500"
+        left="12px"
+        m="0"
+        p="0 4px"
+        pointerEvents="none"
+        position="absolute"
+        top={0}
+        transform={'translateY(-50%)'}
+        zIndex={2}
+      >
+        {label}
+      </FormLabel>
+      <InputGroup>
+        <ChakraInput
+          _focus={{ borderColor: 'dolbyPurple.400', boxShadow: 'none' }}
+          _hover={{ borderColor: 'dolbyNeutral.500', boxShadow: 'none' }}
+          bg="dolbyNeutral.800"
+          border="2px solid"
+          borderColor="dolbyNeutral.500"
+          borderRadius="8px"
+          fontSize="14px"
+          fontWeight="500"
+          height="48px"
+          onChange={handleChange}
+          placeholder={placeholder}
+          px={3}
+          py={5}
+          textAlign="left"
+          value={value}
+          w="100%"
+        />
+        {value ? (
```

Comment:
We can remove `null` here by using `value && (...`
And let's use `undefined` instead of `null` when necessary

Reply:
Good share, let's keep this.  But should we use `undefined` instead of `null` as typescript pro it and react18 allow this?

Reply:
I think it comes down to semantics. `null` means a value is empty because it was intentionally set to have no value. On the other hand, `undefined` means a value is empty because it wasn't assigned. In this situation, we want to intentionally resolve this expression to be an empty value (i.e. React node), so I think `null` is more appropriate. 

Functionally there isn't any difference, both `null` and `undefined` are supported as you mentioned. Let me know what your thoughts on this are.

Reply:
Check this https://stackoverflow.com/questions/44536340/typescript-what-is-the-difference-between-null-and-undefined#:~:text=You%20may%20use%20either%2C%20just,not%20null%20%3A%20Typescript%20Project%20Styleguide.  

They are same functionally like what you mentioned, but since we are using it in typescript, let's keep the consistency.